js/web/lib/wasm/session-handler.ts

```diff
@@ -31,13 +34,30 @@ export class OnnxruntimeWebAssemblySessionHandler implements SessionHandler {
   inputNames: string[];
   outputNames: string[];
 
-  async loadModel(model: Uint8Array, options?: InferenceSession.SessionOptions): Promise<void> {
+  async loadModel(pathOrBuffer: string|Uint8Array, options?: InferenceSession.SessionOptions): Promise<void> {
     if (!ortInit) {
       await initOrt(env.wasm.numThreads!, getLogLevel(env.logLevel!));
       ortInit = true;
     }
 
-    [this.sessionId, this.inputNames, this.outputNames] = await createSession(model, options);
+    if (typeof pathOrBuffer === 'string') {
+      if (typeof fetch === 'undefined') {
+        // node
+        let model = await promisify(readFile)(pathOrBuffer);
+        [this.sessionId, this.inputNames, this.outputNames] = await createSession(model, options);
+      } else {
+        // browser
+        let modelData : [number, number];
+        {
+          const response = await fetch(pathOrBuffer);
```

Comment:
everything related to the fetch() is in this closure and should be collectable after we exit it but I don't see it being collected.

Reply:
the boundary probably be function level instead of block level inside a function.

things may be good if do like this:
```ts

if (typeof pathOrBuffer === 'string') {

  await doFirstStepFetchAndCreateSessionAllocate(pathOrBuffer);

  // the engine should be happy at this point to GC the resources
  await createSession(...);
}
```

Reply:
yeap, I tried that but did not free up the response array either.